packages/components/src/components/as-category-widget/as-category-widget.tsx

```diff
@@ -0,0 +1,195 @@
+
+  @State() private selectedCategories: string[] = [];
+  @State() private numberOfVisibleCategories: number = 5;
+
+  @Method()
+  public getSelectedCategories() {
+    return this.selectedCategories;
+  }
+
+  @Method()
+  public clearSelection() {
+    this.selectedCategories = [];
+    this._onCategoriesChanged();
+  }
+
+  public render() {
+    return [
+      this._renderHeader(),
+      <ul class='as-category-widget__list'>{this._renderCategories()}</ul>,
+      this._renderFooter()
+    ];
+  }
+
+  private _renderHeader() {
+    if (!this.showHeader) {
+      return;
+    }
+
+    return [
+      <h2 class='as-subheader'>{this.heading}</h2>,
+      <p class='as-category-widget__description as-body'>{this.description}</p>,
+    ];
+  }
+
+  private _renderCategories() {
+    let otherCategoryTemplate;
+    const moreCategoriesThanVisible = this.categories.length > this.numberOfVisibleCategories;
+    const { categories, otherCategory } = this.parseCategories();
+
+    const categoriesToRender =  moreCategoriesThanVisible
+      ? categories.slice(0, this.numberOfVisibleCategories)
+      : categories;
+
+    const maximumValue = this.useTotalPercentage
+      ? this._getCategoriesTotalValue(this.categories)
+      : this._getVisibleCategoriesMaximumValue();
+
+    if (otherCategory || moreCategoriesThanVisible) {
+      otherCategoryTemplate = this._renderOtherCategory(otherCategory, { maximumValue });
+    }
+
+    return [
+      categoriesToRender.map((category: Category) => this._renderCategory(category, { maximumValue })),
+      otherCategoryTemplate
+    ];
+  }
+
+  private _renderCategory(category: Category, options: CategoryOptions) {
+    const isSelected = this._isSelected(category.name);
+    const isAnyCategorySelected = this.selectedCategories.length > 0;
+    const barColor = !options.isOther
+      ? this._getBarColor(category.color || this.defaultBarColor, { isSelected })
+      : OTHER_CATEGORY_COLOR;
+
+    const progressStyles = {
+      backgroundColor: barColor,
+      width: `${(category.value / options.maximumValue) * 100}%`
+    };
+
+    const cssClasses = {
+      'as-category-widget__category': true,
+      'as-category-widget__category--not-selected': isAnyCategorySelected && (!isSelected || options.isOther),
+      'as-category-widget__category--other': options.isOther,
+      'as-category-widget__category--selected': isSelected
+    };
+
+    return (
+      <li class={cssClasses} onClick={() => this._toggleCategory(category)}>
+        <p class='as-category-widget__category-title as-body' data-value={readableNumber(category.value)}>
+          {category.name}
+        </p>
+
+        <div class='as-category-widget__bar'>
+          <div class='as-category-widget__value' style={progressStyles}></div>
+        </div>
+      </li>
+    );
+  }
+
+  private _renderOtherCategory(category: Category, options: CategoryOptions) {
+    const categoryData = category || {
```

Comment:
since typescript supports default parameters we could use them, but the syntax is not very familiar in this case:

```ts
private _renderOtherCategory(
    category: Category = {
      name: 'Other',
      value: this._getCategoriesTotalValue(
        this.categories.slice(this.numberOfVisibleCategories, this.categories.length)
      )
    },
    options: CategoryOptions) {

    return this._renderCategory(category, { maximumValue: options.maximumValue, isOther: true });
  }
```

Reply:
No, I think it's easier to follow as it is right now. That syntax looks weird.